app/Http/Controllers/Api/Auth/OAuthController.php

```diff
@@ -129,26 +128,29 @@ private function validateRequest(Request $request)
      * Proxy a request to the OAuth server.
      *
      * @param array $data the data to send to the server
+     *
      * @return array
+     * @throws \Safe\Exceptions\JsonException
      */
     private function proxy(array $data = [])
     {
-        $http = new Client([
-            'timeout' => 20,
-        ]);
-        $url = App::runningUnitTests() ? config('app.url').'/oauth/token' : route('passport.token');
-        $response = $http->post($url, [
-            'form_params' => [
-                'grant_type' => $data['grantType'],
-                'client_id' => config('monica.mobile_client_id'),
-                'client_secret' => config('monica.mobile_client_secret'),
-                'username' => $data['username'],
-                'password' => $data['password'],
-                'scope' => '',
-            ],
-        ]);
-
-        $data = json_decode($response->getBody());
+        $http = app(\Illuminate\Contracts\Http\Kernel::class);
+        $response = $http->handle(
+            \Illuminate\Http\Request::create(
+                '/oauth/token',
```

Comment:
Could you use the `$url` var define before? I prefer to use `route` when it's possible, and here I'm not sure if it's working with subdirectory installations.

Reply:
Sorry, I can't understand what is wrong with the style ci. Can you help me on it ?